services/skus/credentials.go

```diff
@@ -720,7 +720,20 @@ func (s *Service) DeleteOrderCreds(ctx context.Context, orderID uuid.UUID, isSig
 	}
 
 	if doTlv2 {
-		if err := s.Datastore.DeleteTimeLimitedV2OrderCredsByOrderTx(ctx, tx, orderID); err != nil {
+		var itemIDs []*uuid.UUID
+		if reqID == nil {
+			// legacy, so put all of the orders' items in a list to be hard deleted
+			for _, v := range order.Items {
+				itemIDs = append(itemIDs, &v.ID)
```

Comment:
Will this not result in the same value for every item given its taking a pointer in the loop?

Reply:
Also, it's unclear why use a slice of pointers to uuids. Under the hood, a UUID is a small array (not even a slice), which is OK to pass around by value, and `uuid.Nil` serves as a way to pass/indicate emptiness.

Additionally, `itemIDs` can be preallocated as we know the number of items upfront.

As for the `reqID`, the check should become `if uuid.Equal(reqID, uuid.Nil)`.

Reply:
Updated by the patch #2314.